cli/tests/test_status_command.rs

```diff
@@ -62,3 +62,53 @@ fn test_status_ignored_gitignore() {
     Parent commit: zzzzzzzz 00000000 (empty) (no description set)
     "###);
 }
+
+// See <https://github.com/martinvonz/jj/issues/3108>
+#[test]
+fn test_status_display_rebase_instructions() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
+
+    let repo_path = test_env.env_root().join("repo");
+    let conflicted_path = repo_path.join("conflicted.txt");
+
+    // PARENT: Write the initial file
+    std::fs::write(&conflicted_path, "initial contents").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["describe", "--message", "Initial contents"]);
+
+    // CHILD1: New commit on top of <PARENT>
+    test_env.jj_cmd_ok(
+        &repo_path,
+        &["new", "--message", "First part of conflicting change"],
+    );
+    std::fs::write(&conflicted_path, "Child 1").unwrap();
+
+    // CHILD2: New commit also on top of <PARENT>
+    test_env.jj_cmd_ok(
+        &repo_path,
+        &[
+            "new",
+            "--message",
+            "Second part of conflicting change",
+            "@-",
+        ],
+    );
+    std::fs::write(&conflicted_path, "Child 2").unwrap();
+
+    // CONFLICT: New commit that is conflicted by merging <CHILD1> and <CHILD2>
+    test_env.jj_cmd_ok(&repo_path, &["new", "--message", "boom", "all:(@-)+"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["status"]);
+
+    insta::assert_snapshot!(stdout, @r###"
+    The working copy is clean
+    There are unresolved conflicts at these paths:
+    conflicted.txt    2-sided conflict
+    To resolve the conflicts, either work on the current change or create a new one to squash later.
+    Then use `jj resolve`, or edit the conflict markers in the file directly.
+    Once the conflicts are resolved, you may want inspect the result with `jj diff`.
+    Then run `jj squash` to move the resolution into the conflicted commit.
```

Comment:
I think the last two lines are only relevant if the user creates a new change, which can be a bit confusing.